### DISCUSSION TITLE
What metrics does compreface use to compare face similarity?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [exadel-inc/CompreFace#864] [Q&A]
What metrics does compreface use to compare face similarity?
Does it use Euclidean distance (L2)? Or something different? 

How can I improve search results using milvus?
Is it better to use 
Euclidean distance (L2)
or
Inner product (IP)
?

Maybe somehow change vectors(normalize or something else) before inserting to milvus?

What distance metric algorithm  compreface uses to compare two faces?
Does it somehow changes vectors?

**Comment by martinenkoEduard** [comment]
Do I need to use something like  faiss.normalize_L2(normalized)

**Reply by pospielov** [reply]
could you describe what `faiss.normalize_L2(normalized)` function does?

**Comment by pospielov** [comment]
I answered it here:
https://github.com/exadel-inc/CompreFace/discussions/819#discussioncomment-3454811
